ee/modules/610-service-with-healthchecks/crds/servicewithhealthchecks.yaml

```diff
@@ -0,0 +1,785 @@
+                          format: int32
+                          type: integer
+                          minimum: 1
+                          default: 1
+                          description: |-
+                            Minimum consecutive successes for the probe to be considered successful after having failed.
+                        mode:
+                          enum:
+                          - HTTP
+                          - TCP
+                          - PostgreSQL
+                          type: string
+                          description: |-
+                            The mode for the probe.
+                        http:
+                          properties:
+                            host:
+                              description: |-
+                                Hostname to connect to, defaults to the pod's IP address.
+                              type: string
+                              format: hostname
+                            httpHeaders:
+                              description: Custom headers that will be added to the HTTP request and used when
+                                performing the probe. Duplicate headers are
+                                allowed.
+                              items:
+                                description:
+                                properties:
+                                  name:
+                                    description: |-
+                                      The header name.
+
+                                      This will be canonicalized upon output, so case-variant names will be understood as the same header.
+                                    type: string
+                                  value:
+                                    description: The header value.
+                                    type: string
+                                required:
+                                - name
+                                - value
+                                type: object
+                              type: array
+                            method:
+                              description: Method to use for the request.
+                              type: string
+                              default: GET
+                              enum:
+                              - GET
+                              - PUT
+                              - POST
+                              - HEAD
+                            path:
+                              description: Path to access on the HTTP server.
+                              type: string
+                            scheme:
+                              description: |-
+                                Scheme to use for connecting to the host.
+                              type: string
+                              default: HTTP
+                              enum:
+                              - HTTP
+                              - HTTPS
```

Comment:
For HTTPS, there has to be an option to provide a CA.

Reply:
This is also odd that http probe can be https.

Reply:
The HTTPS was stolen from [k8s probe references](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#httpgetaction-v1-core).

Reply:
We will forbid using POST requests in probes.